git_machete/git_operations.py

```diff
@@ -64,7 +64,7 @@ def get_default_editor(self) -> Optional[str]:
         proposed_editor_funs: List[Tuple[str, Callable[[], Optional[str]]]] = [
             ("$" + git_machete_editor_var, lambda: os.environ.get(git_machete_editor_var)),
```

Comment:
https://github.com/VirtusLab/git-machete/issues/251 ... something to consider while you're at this PR. Not to be done right now, but just to confirm that the existing `run_git` and `popen_git` calls from client.py are safe from the PoV of cache validity.

Reply:
I looked into it and they're not. For example `delete_unmanaged` in my opinion should be flushing the cache. Cache clearing should be done on the side of the actual executor of the command that invalidates the cache. `delete_unmanaged` is defined in `client.py` and does that amongst other lines: 
```
if ans in ('y', 'yes'):
    self.__git.run_git("branch", "-d" if is_merged_to_remote else "-D", branch)
```
Ideally we would have here a call to a wrapper around `git branch -d/-D,` and this wrapper would call `run_git` and also invalidate/clear the cache. I would say that the issue #251 should be pulled into `v3.4.0 `release and as a followup/part of this story cache management should be reconsidered.

Reply:
Okay! into v3.4.0 it goes then